common/src/test/java/gov/cms/ab2d/common/health/DatabaseAvailableTest.java

```diff
@@ -0,0 +1,316 @@
+package gov.cms.ab2d.common.health;
+
+import gov.cms.ab2d.common.SpringBootApp;
+import gov.cms.ab2d.common.util.AB2DPostgresqlContainer;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.TestPropertySource;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import javax.sql.DataSource;
+
+import java.sql.*;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.Executor;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.anyString;
+
+@SpringBootTest(classes = SpringBootApp.class)
+@TestPropertySource(locations = "/application.common.properties")
+@Testcontainers
+class DatabaseAvailableTest {
+    @Autowired
+    private DataSource dataSource;
+    @Mock
+    private DataSource bogusDS;
+
+    @Container
+    private static final PostgreSQLContainer postgreSQLContainer= new AB2DPostgresqlContainer();
+
+    @Test
+    public void testDatasource() throws SQLException {
+        assertTrue(DatabaseAvailable.isDbAvailable(dataSource));
+        assertFalse(DatabaseAvailable.isDbAvailable(null));
+        Mockito.when(bogusDS.getConnection()).thenReturn(null);
+        assertFalse(DatabaseAvailable.isDbAvailable(bogusDS));
+        Mockito.when(bogusDS.getConnection()).thenReturn(new Connection() {
```

Comment:
Hmm. Curious why we're having to do this. You could just create a `Connection` mock. And in case you need one (or more) of its methods to actually do _non-mock_ stuff (an action or something), Mockito has API for that as well.

Reply:
Lazy. When I entered new Connection, it auto-generated the methods that you have overwrite and it worked so I didn't bother changing it figuring I could fill the overwritten methods with other stuff later if I wanted better testing. I'll refactor.